app/src/main/java/com/example/simplemp3/ui/activity/MainActivity.kt

```diff
@@ -1,13 +1,106 @@
 package com.example.simplemp3.ui.activity
 
-import androidx.appcompat.app.AppCompatActivity
+import android.content.ComponentName
+import android.content.Context
+import android.content.Intent
+import android.content.ServiceConnection
 import android.os.Bundle
+import android.os.Handler
+import android.os.IBinder
+import android.widget.SeekBar
+import androidx.appcompat.app.AppCompatActivity
 import com.example.simplemp3.R
+import com.example.simplemp3.service.PlayService
+import com.example.simplemp3.service.PlayService.Companion.DEFAULT_LOOP_STATE
+import kotlinx.android.synthetic.main.activity_main.*
+
 
 class MainActivity : AppCompatActivity() {
 
+    companion object {
+        const val DELAY_SEEK_TIME: Long = 100
+        const val SONG_NAME = "Ao Moi Ca Mau"
+        const val ARTIST_NAME = "Jang Mi"
+
+        fun getPlayerIntent(context: Context) = Intent(context, PlayService::class.java)
+    }
+
+    private var isServiceBound = false
+    private var isPlaying = false
+    private var isLooping = DEFAULT_LOOP_STATE
+    private var playService: PlayService? = null
+    private val playServiceConnection = object : ServiceConnection {
+        override fun onServiceConnected(name: ComponentName?, service: IBinder?) {
+            val binder = service as PlayService.PlayerServiceBinder
+            playService = binder.getService()
+            isServiceBound = true
+        }
+
+        override fun onServiceDisconnected(name: ComponentName?) {
+            isServiceBound = false
+        }
+    }
+
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
         setContentView(R.layout.activity_main)
+        bindService(getPlayerIntent(this), playServiceConnection, BIND_AUTO_CREATE)
+        changeLoopState()
+        setDefaultSongDetails()
+        setHandlers()
+        setListeners()
+    }
+
+    private fun setDefaultSongDetails() {
```

Comment:
```suggestion
    private fun displaySongDetail(songName : String = SONG_NAME, artist:String = ARTIST_NAME) {
```
em nên sử dụng fun với default value ntn thì đúng hơn